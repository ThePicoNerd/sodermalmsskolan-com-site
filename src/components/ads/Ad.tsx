import React, { FunctionComponent, useEffect } from "react";
import { ADSENSE_CLIENT, ADSENSE_SLOT } from "../../lib/adsense/constants";

export type AdFormat = "auto" | "fluid";

export type AdLayout = "in-article";

export interface AdProps {
  className?: string;
  format?: AdFormat,
  layoutKey?: string,
  layout?: string,
  fullWidthResponsive?: boolean,
}

/**
 * Google AdSense ad.
 *
 * @param {React.PropsWithChildren<AdProps>} props Props.
 *
 * @returns {React.ReactElement} The rendered ad.
 */
const Ad: FunctionComponent<AdProps> = ({
  className = "adsbygoogle",
  format = "auto",
  layoutKey,
  layout,
  fullWidthResponsive,
}) => {
  useEffect(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    ((window as any).adsbygoogle = (window as any).adsbygoogle || []).push({});
  }, []);

  return (
    <ins
      className={className}
      data-ad-format={format}
      data-ad-layout-key={layoutKey}
      data-ad-layout={layout}
      data-full-width-responsive={fullWidthResponsive ? "true" : undefined}
      data-ad-client={ADSENSE_CLIENT}
      data-ad-slot={ADSENSE_SLOT}
    />
  );
};

export default Ad;
