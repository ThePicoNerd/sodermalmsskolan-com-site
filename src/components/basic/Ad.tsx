import React from "react";
import Col from "react-bootstrap/Col";
import Row from "react-bootstrap/Row";
import Container from "react-bootstrap/Container";
import AdSense from "react-adsense";
import { Section } from "./Section";

export const ResponsiveAd: React.FunctionComponent = () => {
  return (
    <AdSense.Google
      client={process.env.adsenseClient}
      slot={process.env.adsenseSlot}
      style={{ display: "block" }}
      layout="in-article"
      format="fluid"
    />
  );
};

export const AdSection: React.FunctionComponent = () => {
  return (
    <Section>
      <Container>
        <Row>
          <Col>
            <ResponsiveAd />
          </Col>
        </Row>
      </Container>
    </Section>
  );
};
