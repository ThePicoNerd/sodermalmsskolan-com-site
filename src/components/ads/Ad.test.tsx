import React from "react";
import { render } from "@testing-library/react";
import Ad from "./Ad";

test("<Ad />", () => {
  const result = render(<Ad />);

  expect(result.asFragment()).toMatchSnapshot();
});
