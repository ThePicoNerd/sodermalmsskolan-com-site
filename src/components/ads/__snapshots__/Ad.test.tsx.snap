// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ad /> 1`] = `
<DocumentFragment>
  <ins
    class="adsbygoogle"
    data-ad-client="ca-pub-1944185766034883"
    data-ad-format="auto"
    data-ad-slot="5909725353"
  />
</DocumentFragment>
`;
